docs/reference/coding-guidelines.md

```diff
@@ -98,6 +98,58 @@ export {
 } from './internalApi.js';
 ```
 
+### Variable Naming
+
+Gutenberg inherits [WordPress' naming conventions of camel-casing](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#naming-conventions):
+
+>Variable and function names should be full words, using camel case with a lowercase first letter. This is an area where this standard differs from the WordPress PHP coding standards.
+>
+>Constructors intended for use with `new` should have a capital first letter (UpperCamelCase).
+
+However, Gutenberg is more specific about its handling of abbreviations, acronyms, constants, and the ES2015 class construct.
+
+#### Abbreviations and Acronyms
+
+[*Abbreviations*](https://en.wikipedia.org/wiki/Abbreviation) must be written as camel case, with an initial capitalized letter followed by lowercase letters.
+
+[*Acronyms*](https://en.wikipedia.org/wiki/Acronym) must be written with each of its composing letters capitalized. This is intended to reflect that each letter of the acronym is a proper word in its expanded form.
+
+**Examples:**
+
+```js
+// "Id" is an abbreviation of "Identifier":
+const userId = 1;
+
+// "DOM" is an acronym of "Document Object Model":
+const currentDOMDocument = window.document;
+```
+
+#### Class Definition
+
+A [`class` definition](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) must use the UpperCamelCase convention, regardless of whether it is intended to be used with `new` construction.
```

Comment:
Should we note that React function components should be UpperCamelCase as well?

Reply:
Just noting that we aren't consistent ourselves because we use `edit` and `save` for blocks where we should really be using `Edit` and `Save`...

Reply:
It's a tricky one. I think the functional components should be UpperCamelCase, yes. Part of that is an expectation that it could be converted to a class component at any point. And also just for consistency. `edit` and `save` are interesting to note. They were originally meant to be "just functions", not strictly so much a component: `save` at one time could even just return a markup string. When we extract the component proper, we do name them "correctly" (e.g. `ImageEdit`). And in general a capitalized property name seems a bit strange, though I think at one point we were doing the destructure as `const { edit: Edit } = blockType;`. Maybe it's enough to make that distinction on declaration of variable / class, vs. property name?

Reply:
> I think at one point we were doing the destructure as `const { edit: Edit } = blockType;`.  Maybe it's enough to make that distinction on declaration of variable / class, vs. property name?

Yes, it makes sense. In addition we can change the way we import `edit` for core blocks to work as you described. I can do it as a follow-up as I introduced this shortcut 😓